js/GroupedBarChart.js

```diff
@@ -174,26 +195,161 @@ var GroupedBarChart = function(param)
               .attr("y", that.h)
               .attr("width", x1.rangeBand())
               .attr("height", 0)
-              .attr("fill", function(d, i) { return z(i); })
+              .attr("fill", function(d, i) { return c(_.indexOf(data, d)); })
               // The transition produces a bouncing effect for the bar
               .transition()
                 // Each bar will be delayed depending on its position in the graph
                 .delay(function(d, i) { return  _.indexOf(data, d) * 50;} )
-                .duration(150)
+                .duration(barsAnimationTime/3)
                 // Expand height first (bounce effect)
                 .attr('y', function(d) { return y(d.yval) - 50; })
-                .attr('height', function(d) { return (that.h - y(d.yval)) + 50 })
+                .attr('height', function(d) { return (that.h - y(d.yval)) + 50 ;})
                   .transition()
-                  .duration(150)
+                  .duration(barsAnimationTime/3)
                   // Lower the height after (bounce effect)
                   .attr('y', function(d) { return y(d.yval) + 15; })
                   .attr('height', function(d) { return that.h - y(d.yval) - 15; })
                     .transition()
-                    .duration(150)
+                    .duration(barsAnimationTime/3)
                     // Turn back to original height
                     .attr('y', function(d) { return y(d.yval); })
-                    .attr('height', function(d) { return that.h - y(d.yval); })
+                    .attr('height', function(d) { return that.h - y(d.yval); });
+
+
+                    /*Data point funcitons*/
+                    var c_mouseover = function(d){
+                      d3.select(this)
+                          .classed('hover', true)
+                          .transition()
+                          .duration(400)
+                          .attr('r', 5 * 1.5)
+                          .transition()
+                          .duration(150)
+                          .attr('r', 5 * 1.25);
+                    };
+
+                    var c_mouseout = function(d){
+                      d3.select(this)
+                          .classed('hover', false)
+                          .transition()
+                          .duration(150)
+                          .attr('r', 5);
+                    };
+
+                    that.addLine = d3.svg.line()
+                        .x(
+                            function(d)
+                            {
+                                return x0(d.x0) + x1(d.x1) + x1.rangeBand()/2 ;
+                            })
+                        .y(
+                            function(d)
+                            {
+                                return y(d.y);
+                            });
+
+                    function createLineData(arrY, arrX0, arrX1){
```

Comment:
Yes that functionality can be in update chart, probably  in a  better way or a way that doesn't involve the function. However, the sources are three completely different arrays,  so mapping one array where the name equals total average won't work. If you test it with  this one line  function you'll find that the circles and line Will not render. I agree it should be revisited